client/src/models/Report.js

```diff
@@ -410,4 +410,62 @@ export default class Report extends Model {
     })
     return { taskToAssessmentUuid, taskAssessments }
   }
+
+  hasConflict(that) {
+    if (this.uuid === that.uuid) {
+      return false // same report is not a conflicting report
+    }
+
+    let thisStart
+    let thisEnd
+
+    if (this.duration) {
+      thisStart = moment(this.engagementDate)
+      thisEnd = moment(this.engagementDate).add(this.duration, "minute")
+    } else {
+      // It is an all-day event if duration is not given
```

Comment:
Strictly speaking, it is an all-day event when `!Settings.engagementsIncludeTimeAndDuration || this.duration === null`.